src/logic/LocalMessageCache.tsx

```diff
@@ -65,14 +65,33 @@ const readQueue = (slotName: MessageSlotName): CachedMessage[] => {
 
 const hydrateMessage = (
     messageData: MessageData,
-    appboyInstance: typeof appboy,
-): appboy.HtmlMessage => {
+    brazeInstance: typeof braze,
+): braze.HtmlMessage => {
+    /**
+     * Removed APIs
+     * The following methods and properties have been removed:
+     * "cardId" and "campaignId" properties of In-App Messages
+     * https://github.com/braze-inc/braze-web-sdk/blob/master/UPGRADE_GUIDE.md#removed-apis-1
+     * We will move them into "extras" to keep the same data as before
+     */
+    let extras: Record<string, string> | undefined = messageData.extras;
+    if (messageData.cardId) {
+        if (!extras) {
+            extras = {};
+        }
+        extras.cardId = messageData.cardId;
+    }
+    if (messageData.campaignId) {
+        if (!extras) {
+            extras = {};
+        }
+        extras.campaignId = messageData.campaignId;
+    }
```

Comment:
I'm not really a TS programmer but for simplicity is it worth doing something along these lines?
```suggestion
    const extras =  { 
      ...messageData.extras || {},
      ...messageData.cardId && { cardId: messageData.cardId },
      ...messageData.campaignId && { campaignId: messageData.campaignId }
    }
```

Reply:
Hey @johnduffell , thanks for your suggestion! That was my first approach. But the thing is, by doing `...messageData.extras || {},` we are taking a decision of initializing "extras" even if we don't need to. I was afraid of doing that thinking about the consequences, but since you also believe that it's ok, I will change it, thanks!

Reply:
see what you mean, I didn't test that scenario.  I would have thought it would be ok.

Trying to handle it in TS turns into something that although shorter than the original it doesn't seem clear any more
```
    const extrasArray = [ 
      messageData.extras,
      messageData.cardId && { cardId: messageData.cardId },
      messageData.campaignId && { campaignId: messageData.campaignId }
    ].filter((v) => v !== undefined);

    const extras = extrasArray.length > 0 ? Object.assign({}, ...extrasArray) : undefined;
```

also see https://playcode.io/2322453